Linter ignores multiline `/* eslint-env */` directives

<!--
    ESLint adheres to the Open JS Foundation Code of Conduct:
    https://eslint.org/conduct

    This template is for bug reports. If you are here for another reason, please see below:

    1. To propose a new rule: https://eslint.org/docs/developer-guide/contributing/new-rules
    2. To request a rule change: https://eslint.org/docs/developer-guide/contributing/rule-changes
    3. To request a change that is not a bug fix, rule change, or new rule: https://eslint.org/docs/developer-guide/contributing/changes
    4. If you have any questions, please stop by our chatroom: https://eslint.org/chat/help

    Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->


**Tell us about your environment**

<!--
    If you are using ESLint v6.5.0 or later, you can run ESLint with the `--env-info` flag and paste the output here.
-->

* **ESLint Version:** v7.27.0
* **Node Version:** v12.18.4
* **npm Version:** v6.14.6
* **Operating System:** windows

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default

<!-- if you are using a parser other than ESLint's default, please try using the default parser to verify if the issue is parser-specific -->

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

<!-- Paste your configuration below: -->
```js
module.exports = {
  rules: {
    "no-undef": "error"
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

<!-- Paste the source code below: -->
```js
/* eslint-env browser 
*/

window;

```

<!-- Paste the command you used to run ESLint: -->
```bash
npx eslint index.js
```

**What did you expect to happen?**

no `no-undef` errors, because `browser` environment enables `window` global

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

```
  4:1  error  'window' is not defined  no-undef
```


**Steps to reproduce this issue:**

<!-- Please tell us exactly how to see the issue you're describing -->

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50LWVudiBicm93c2VyIFxuKi9cblxud2luZG93OyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibm8tdW5kZWYiOjJ9LCJlbnYiOnt9fX0=)

**Are you willing to submit a pull request to fix this bug?**

Yes, if it is a bug.

The issue is: linter ignores _multiline_ `/* eslint-env ... */` comments. A single-line comment works well, as in [this demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50LWVudiBicm93c2VyICovXG5cbndpbmRvdzsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7Im5vLXVuZGVmIjoyfSwiZW52Ijp7fX19).

I couldn't find a test case or a discussion that would confirm that this was intentional. It looks like a common bug with `.` in a regular expression that doesn't have the `s` flag.

Also, `noInlineConfig: true` option does report multiline `eslint-env` as a directive:

```js
module.exports = {
  rules: {
    "no-undef": "error"
  },
  noInlineConfig: true
};
```

```js
/* eslint-env browser 
*/

window;

```

```
  1:1  warning  '/*eslint-env*/' has no effect because you have 'noInlineConfig' setting in your config (.eslintrc.js)
  4:1  error    'window' is not defined  
```

--- a/lib/linter/linter.js
+++ b/lib/linter/linter.js
@@ -444,7 +444,7 @@ function normalizeEcmaVersion(ecmaVersion) {
     return ecmaVersion >= 2015 ? ecmaVersion - 2009 : ecmaVersion;
 }
 
-const eslintEnvPattern = /\/\*\s*eslint-env\s(.+?)\*\//gu;
+const eslintEnvPattern = /\/\*\s*eslint-env\s([\s\S]+?)\*\//gu;
 
 /**
  * Checks whether or not there is a comment which has "eslint-env *" in a given text.

--- a/lib/linter/linter.js
+++ b/lib/linter/linter.js
@@ -444,7 +444,7 @@ function normalizeEcmaVersion(ecmaVersion) {
     return ecmaVersion >= 2015 ? ecmaVersion - 2009 : ecmaVersion;
 }
 
-const eslintEnvPattern = /\/\*\s*eslint-env\s([\s\S]+?)\*\//gu;
+const eslintEnvPattern = /\/\*\s*eslint-env\s(.+?)\*\//gsu;
 
 /**
  * Checks whether or not there is a comment which has "eslint-env *" in a given text.

--- a/tests/lib/linter/linter.js
+++ b/tests/lib/linter/linter.js
@@ -2957,6 +2957,18 @@ var a = "test2";
             assert.strictEqual(messages.length, 0);
         });
 
+        // https://github.com/eslint/eslint/issues/14652
+        it("should not report a violation", () => {
+            const code = "/*eslint-env es6\n */ new Promise();";
+
+            const config = { rules: { "no-undef": 1 } };
+
+            const messages = linter.verify(code, config, filename);
+
+            assert.strictEqual(messages.length, 0);
+        });
+
+
         it("should not report a violation", () => {
             const code = `/*${ESLINT_ENV} mocha,node */ require();describe();`;
 

--- a/tests/lib/linter/linter.js
+++ b/tests/lib/linter/linter.js
@@ -2959,15 +2959,20 @@ var a = "test2";
 
         // https://github.com/eslint/eslint/issues/14652
         it("should not report a violation", () => {
-            const code = "/*eslint-env es6\n */ new Promise();";
-
+            const codes = [
+                "/*eslint-env es6\n */ new Promise();",
+                "/*eslint-env browser,\nes6 */ window;Promise;",
+                "/*eslint-env\nbrowser,es6 */ window;Promise;"
+            ];
             const config = { rules: { "no-undef": 1 } };
 
-            const messages = linter.verify(code, config, filename);
+            for (const code of codes) {
+                const messages = linter.verify(code, config, filename);
 
-            assert.strictEqual(messages.length, 0);
+                assert.strictEqual(messages.length, 0);
+            }
+
         });
-
 
         it("should not report a violation", () => {
             const code = `/*${ESLINT_ENV} mocha,node */ require();describe();`;

